indy-repository-manager/src/main/java/org/jboss/pnc/indyrepositorymanager/IndyRepositorySession.java

```diff
@@ -710,6 +719,95 @@ private String computeIdentifier(final TrackedContentEntryDTO transfer) {
         return identifier;
     }
 
+    /**
+     * Computes purl string for an artifact.
+     *
+     * @param transfer the download or upload that we want to generate identifier for
+     * @return generated purl
+     */
+    private String computePurl(final TrackedContentEntryDTO transfer) {
+        String purl = null;
+
+        try {
+            switch (transfer.getStoreKey().getPackageType()) {
+                case MAVEN_PKG_KEY:
+
+                    ArtifactPathInfo pathInfo = ArtifactPathInfo.parse(transfer.getPath());
+                    if (pathInfo != null) {
+                        // See https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#maven
+                        PackageURLBuilder purlBuilder = PackageURLBuilder.aPackageURL()
+                                .withType(PackageURL.StandardTypes.MAVEN)
+                                .withNamespace(pathInfo.getProjectId().getGroupId())
+                                .withName(pathInfo.getProjectId().getArtifactId())
+                                .withVersion(pathInfo.getVersion())
+                                .withQualifier(
+                                        "type",
+                                        StringUtils.isEmpty(pathInfo.getType()) ? "jar" : pathInfo.getType());
+
+                        if (!StringUtils.isEmpty(pathInfo.getClassifier())) {
+                            purlBuilder.withQualifier("classifier", pathInfo.getClassifier());
+                        }
+                        purl = purlBuilder.build().toString();
+                    }
+                    break;
+
+                case NPM_PKG_KEY:
+
+                    NpmPackagePathInfo npmPathInfo = NpmPackagePathInfo.parse(transfer.getPath());
+                    if (npmPathInfo != null) {
+                        // See https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#npm
+                        PackageURLBuilder purlBuilder = PackageURLBuilder.aPackageURL()
+                                .withType(PackageURL.StandardTypes.NPM)
+                                .withVersion(npmPathInfo.getVersion().toString());
+
+                        String[] scopeAndName = npmPathInfo.getName().split("/");
+                        if (scopeAndName != null && scopeAndName.length > 0) {
```

Comment:
I think there is no way how scopeAndName could be null.

Reply:
right, changing